core/node/da_clients/src/eigen/sdk.rs

```diff
@@ -1,61 +1,132 @@
 
 impl RawEigenClient {
     pub(crate) const BUFFER_SIZE: usize = 1000;
 
-    pub async fn new(
-        rpc_node_url: String,
-        inclusion_polling_interval_ms: u64,
-        private_key: SecretKey,
-    ) -> anyhow::Result<Self> {
+    pub async fn new(private_key: SecretKey, config: EigenConfig) -> anyhow::Result<Self> {
         let endpoint =
-            Endpoint::from_str(rpc_node_url.as_str())?.tls_config(ClientTlsConfig::new())?;
+            Endpoint::from_str(config.disperser_rpc.as_str())?.tls_config(ClientTlsConfig::new())?;
         let client = DisperserClient::connect(endpoint)
             .await
             .map_err(|e| anyhow::anyhow!("Failed to connect to Disperser server: {}", e))?;
-        let polling_interval = Duration::from_millis(inclusion_polling_interval_ms);
-
-        let account_id = get_account_id(&private_key);
 
+        let verifier_config = VerifierConfig {
+            verify_certs: true,
+            rpc_url: config.eigenda_eth_rpc.clone(),
+            svc_manager_addr: config.eigenda_svc_manager_address.clone(),
+            max_blob_size: config.blob_size_limit,
+            path_to_points: config.path_to_points.clone(),
+            eth_confirmation_depth: config.eth_confirmation_depth.max(0) as u32,
+            private_key: hex::encode(private_key.secret_bytes()),
+            chain_id: config.chain_id,
+        };
+        let verifier = Verifier::new(verifier_config)
+            .map_err(|e| anyhow::anyhow!(format!("Failed to create verifier {:?}", e)))?;
         Ok(RawEigenClient {
             client,
-            polling_interval,
             private_key,
-            account_id,
+            config,
+            verifier,
         })
     }
 
-    pub async fn dispatch_blob(&self, data: Vec<u8>) -> anyhow::Result<String> {
+    async fn dispatch_blob_non_authenticated(&self, data: Vec<u8>) -> anyhow::Result<String> {
+        let padded_data = convert_by_padding_empty_byte(&data);
+        let request = disperser::DisperseBlobRequest {
+            data: padded_data,
+            custom_quorum_numbers: vec![],
+            account_id: String::default(), // Account Id is not used in non-authenticated mode
+        };
+
+        let mut client_clone = self.client.clone();
+        let disperse_reply = client_clone.disperse_blob(request).await?.into_inner();
+
+        let disperse_time = Instant::now();
+        let blob_info = self
+            .await_for_inclusion(client_clone, disperse_reply)
+            .await?;
+        let disperse_elapsed = Instant::now() - disperse_time;
+
+        let blob_info = blob_info::BlobInfo::try_from(blob_info)
+            .map_err(|e| anyhow::anyhow!("Failed to convert blob info: {}", e))?;
+        self.verifier
+            .verify_commitment(blob_info.blob_header.commitment.clone(), data)
+            .map_err(|_| anyhow::anyhow!("Failed to verify commitment"))?;
+
+        self.loop_verify_certificate(blob_info.clone(), disperse_elapsed)
```

Comment:
this is blocking the execution until the EigenDA proofs are on L1, which shouldn't be the case
we should be polling for the available state roots on `get_inclusion_data` stage

this way your `blob_id` will be the single identifier, not the serialized struct
and you will use the blob_id down the line to get the inclusion data (and verify that EigenDA batch was proved on L1)